compiler/noirc_frontend/src/elaborator/types.rs

```diff
@@ -1057,6 +1040,55 @@ impl<'context> Elaborator<'context> {
         }
     }
 
+    // Given a unary operator and a type, this method will produce the output type
+    // and a boolean indicating whether to use the trait impl corresponding to the operator
+    // or not. A value of false indicates the caller to use a primitive operation for this
+    // operator, while a true value indicates a user-provided trait impl is required.
+    pub(super) fn prefix_operand_type_rules(
```

Comment:
We should still be able to do this within `type_check_prefix_operand`, I don't think we need two separate functions.

If you want, you can use `try_unify` in the other method to avoid the need to change the match expression to recur on the `rhs_type`.

Reply:
Done! I unified them but kept `prefix_operand_type_rules` instead of `type_check_prefix_operand` to the code looks similar to `elaborate_infix`.

There's one thing that ended up a bit strange, I'll add a comment in that place.